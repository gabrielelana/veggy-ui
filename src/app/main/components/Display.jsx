const Display = ({time, is_shared}) => (
  <div className="column is-three-quarters">
    <div className="box has-text-centered">
      <h1 id="time" className="title is-1">{time}{is_shared && <i className="shared-icon fa fa-share-alt"></i>}</h1>
    </div>
  </div>
)

Display.defaultProps = {
  time: '25:00'
}

export default Display