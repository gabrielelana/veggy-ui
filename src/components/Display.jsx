const React = require('react')

const Display = React.createClass({
  render(){
    return (
      <div className="column is-three-quarters">
        <div className="box has-text-centered">
          <h1 className="title is-1">25:00</h1>
        </div>
      </div>
      )
  }
})

export default Display