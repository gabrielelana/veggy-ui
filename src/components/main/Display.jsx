const React = require('react')

const Display = React.createClass({
  render(){
    return (
      <div className="column is-three-quarters">
        <div className="box has-text-centered">
          <h1 id="timer" className="title is-1">{this.props.timer}</h1>
        </div>
      </div>
      )
  }
})

export default Display