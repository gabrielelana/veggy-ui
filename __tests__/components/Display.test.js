import React from 'react'
import renderer from 'react-test-renderer';
import Display from '../../src/app/main/components/Display'

describe('<Display />', () => {
  
  test('should display the time passed via props', () => {
    const component = renderer.create(<Display time="11:04" />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})