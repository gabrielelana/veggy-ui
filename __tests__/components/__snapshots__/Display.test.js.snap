// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> should display the time passed via props 1`] = `
<div
  className="column is-three-quarters"
>
  <div
    className="box has-text-centered"
  >
    <h1
      className="title is-1"
      id="time"
    >
      11:04
    </h1>
  </div>
</div>
`;
